src/main/java/uk/gov/justice/digital/zone/CuratedZone.java

```diff
@@ -28,9 +25,11 @@ public class CuratedZone implements Zone {
     private final SourceReferenceService sourceReferenceService;
 
     @Inject
-    public CuratedZone(JobArguments jobArguments,
-                       DataStorageService storage,
-                       SourceReferenceService sourceReferenceService) {
+    public CuratedZone(
```

Comment:
CuratedZoneCDC extends CuratedZone is something incorrect here. We should have one base class and extend from one base class rather than applying multi level inheritance

Reply:
Again disagree.

Reply:
The inheritance is logically correct.
`CuratedZoneCDC` is a `CuratedZone` is a `Zone`.
likewise
`CuratedZoneLoad` is a `CuratedZone` is a `Zone`.

As shown in the class diagram below. There are three types of zones (`RawZone`, `StructuredZone` and `CuratedZone`).
The `StructuredZone` and `CuratedZone` have two subtypes based on whether they are for Load or CDC.

```mermaid
classDiagram
RawZone --|> Zone : implements
StructuredZoneLoad --|> StructuredZone : extends
StructuredZoneCDC --|> StructuredZone : extends
StructuredZone --|> Zone : implements
CuratedZoneLoad --|> CuratedZone : extends
CuratedZoneCDC --|> CuratedZone : extends
CuratedZone --|> Zone : implements
```